docker_deface.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+GIVEN_INPUT=$*
+trap own_files ERR INT
+
+own_files()
+{
+    platform=$(echo $GIVEN_INPUT | grep -oP 'system_platform=\K[^&]*')
+    # if UID or GID is present in GIVEN_INPUT collect them and assign to variables
+    if [[ $GIVEN_INPUT == *\-\-user=* ]]
+    then
+        uid=$(echo $GIVEN_INPUT | grep -oP '\-\-user=\K[0-9]*')
+        gid=$(echo $GIVEN_INPUT | grep -oP '\-\-user=[0-9]*:\K[0-9]*')
+    fi
+
+    # dont run any of this if the host system that initiated this container isn't linux as 
```

Comment:
Really? I was not aware of this. I was under the assumption that docker runs everything as root.

Reply:
If you pass docker run `--user <uid>:<gid>` or `--user username` it does a pretty good job of running as a non-root. I think the big issue was that I didn't start from there and had to go back and open up permissions on the docker file system so that the container could run unprivileged.